lib/cli/run-option-metadata.js

```diff
@@ -18,9 +18,11 @@ exports.types = {
     'file',
     'global',
     'ignore',
-    'require',
     'reporter-option',
-    'spec'
+    'require',
+    'spec',
+    'watch-files',
+    'watch-ignore'
```

Comment:
Something is wrong with `loadOptions()` in "cli/options". I have to check the camel-case option of yargs-parser.
When I use `--watch-files <path>` the result is - as it should be - an `array` , when I use `--watchFiles <path>` it's a `string`.

Reply:
Test case:
- input via CLI: `--watchFiles ./path1 --watchFiles ./path2`
- plus input via `.mocharc.yml`: `watchFiles: ['./path3']`

current result: `{watchFiles: [ "./path1", "./path2"]`
Makes sense, since the type of `watchFiles` is `undefined`, not `array`.

Setting `camel-case-expansion` to `true` (in "cli/options"), it's still wrong:
result: `{watch-files: [ "./path1", "./path2"], {watchFiles: [ "./path1", "./path2"]}`
This output seems to be a yargs-parser bug, since an `array`typed option should combine the result.

When I use `--watch-files` instead of `--watchFiles` the output is correct:
result: `{watch-files: [ "./path1", "./path2, "./path3"]}`

So the result is due to our camel-case settings and also due to a yargs-parser bug. I don't want to set a quick shot. Do you have an idea, otherwise let's go on and take care of this later.

Reply:
> Do you have an idea

Unfortunately I don’t. I’m not familiar with  yargs’ configuration loading.

Reply:
I will open a separate issue for this parsing issue.